rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,221 @@
+---
+authors: STeve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+# RFD 67 - Redis database access in AWS ElatiCache and MemoryDB
+
+## Required Approvers
+* Engineering: @r0mant && @smallinsky
+* Product: (@klizhentas || @xinding33)
+* Security: @reed
+
+## What
+
+Support Redis database access in AWS [ElatiCache](https://aws.amazon.com/elasticache/redis/) and [MemoryDB](https://aws.amazon.com/memorydb/).
+
+## Why
+
+Redis is one of the most popular in-memory databases, and Amazon offers AWS managed solutions like ElatiCache and MemoryDB.
+
+As described in [Database Access RFD](https://github.com/gravitational/teleport/blob/master/rfd/0011-database-access.md), mutual TLS is used between database clients and proxy, and between proxy and database service. The same applies for Redis access.
+
+However, neither mutual TLS nor IAM authentication is supported by Redis managed by AWS. This document mainly focus on access from database service to the Redis managed by AWS.
+
+## Details
+
+### In-transit encryption (TLS)
+For security purpose, Teleport should ONLY support databases with in-transit encryption (TLS) enabled.
+
+[Enabling TLS on ElastiCache Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html#in-transit-encryption-enable) can be a complicated process, and all client applications must be updated at the same time. Therefore, Teleport should NOT automatically enable TLS for existing clusters. Redis servers without TLS are NOT supported, and users are recommended to enable TLS separately in order to enable Teleport database access.
+
+For clusters with TLS enabled, Teleport performs full certificate validation at connection.
+
+### Auth
+Redis can be configured with one the following authentication methods, and only one at a time.
+
+#### Auth - Redis with no auth
+Redis can be configured without any auth method. "default" user is always used in this configuration, and no `auth` command is required upon succesfull connection.
+
+#### Auth - Redis with Redis Auth
+[Redis Auth](https://redis.io/commands/auth/) uses a single token/password that must be shared for all client applications.
+
+Teleport does NOT support Redis configured in this state, and users are recommended to migrate the database to use ACL/RBAC.<sup>[1]</sup>
+
+#### Auth - Redis with ACL
+[Redis ACL](https://redis.io/docs/manual/security/acl/) (also known as RBAC for ElatiCache Redis) is introduced in Redis 6. Both ElatiCache and MemoryDB provide APIs to manage database users and user groups for ACL.
+
+As with all other databases, the database users should be created by "database admins" (by users, not Teleport) to assign desired permissions.
+
+When an user runs `tsh db connect --db-user <database-user> <aws-rediaws-redis-server>`, database agent automatically sends `auth <database-user> <password>` to Redis server upon succesfull connection, to prevent user from entering the password manually.
```

Comment:
```suggestion
When a user runs `tsh db connect --db-user <database-user> <aws-redis-server>`, database agent automatically sends `auth <database-user> <password>` to Redis server upon a successful connection, to prevent user from entering the password manually.
```